lib/Backend.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Backend
+  ( BackendEffect (..), readSecret, writeSecret, listSecrets, deleteSecret
+  , Backend (..)
+  , BackendPacked (..)
+  )
+where
+
+import qualified Data.Text           as T
+import qualified Entry               as E
+import qualified Toml
+import qualified Data.HashMap.Strict as HS
+
+import Error                         (CofferError)
+import Polysemy.Error                (Error)
+import Toml                          (TomlCodec)
+
+import Polysemy
+
+-- @TODO - rename Secret to Entry?
+data BackendEffect m a where
+  WriteSecret  :: E.Entry -> BackendEffect m Int
+  ReadSecret   :: [T.Text] -> Maybe Int -> BackendEffect m (Maybe E.Entry)
```

Comment:
The `Int` in `WriteSecret` and the `Maybe Int` in `ReadSecret` are version numbers specific to the vault implementation, right?
But we're not handling versioning, and other `BackendEffect` implementations might not support versioning at all. So maybe these two operations should not take/return version numbers?

Reply:
Done